airflow/ui/src/components/Graph/DownloadButton.tsx

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Button } from "@chakra-ui/react";
+import { Panel, useReactFlow, getNodesBounds, getViewportForBounds } from "@xyflow/react";
+import { toPng } from "html-to-image";
+import { useEffect, useState } from "react";
+import { FiDownload } from "react-icons/fi";
+
+export const DownloadButton = () => {
+  const { getNodes } = useReactFlow();
+  const [dimensions, setDimensions] = useState({ height: 768, width: 1024 });
+
+  useEffect(() => {
+    const updateDimensions = () => {
+      const container = document.querySelector(".react-flow__viewport");
+
+      if (container) {
+        setDimensions({
+          height: container.clientHeight,
+          width: container.clientWidth,
+        });
+      }
+    };
+
+    updateDimensions();
+    globalThis.addEventListener("resize", updateDimensions);
+
+    return () => globalThis.removeEventListener("resize", updateDimensions);
+  }, []);
```

Comment:
Why do we need to keep the dimensions in state? I feel like we can just fetch the dimensions when a user clicks "download"?

Reply:
+1, looks like we can get rid of the useEffect.